### PR TITLE
fix(button): use react button props

### PR DESCRIPTION
**Pull request by bddjong**
In the button refactoring, prop forwarding was lost when MUI was removed from the component. This PR restores its function.

Additionally, it uses the default React interface for HTML Button attributes. This means we don't have to manually support all the default HTML attributes such as `id`, `disabled`, `type` and more!

### CODE REVIEW

**Comment by mjcarsjens** [comment]
@joostvanviegen to the rescue

**Comment by mjcarsjens** [comment]
guess not :(

File: components/Button/src/index.tsx [hunk 1]
```diff
@@ -44,17 +33,22 @@ export interface ButtonProps extends BaseProps {
...
+      'denhaag-button--end-icon': iconAlign === 'end',
+    },
+    props.className,
+  );
+
   const iconWrapped = props.icon !== undefined ? <span className={'denhaag-button__icon'}>{props.icon}</span> : '';
 
   return (
-    <button className={rootClassNames} onClick={props.onClick} disabled={props.disabled} type={props.type}>
+    <button {...props} type={type} className={className} onClick={props.onClick}>
```
Comment (mjcarsjens) [inline components/Button/src/index.tsx:51]: Is it necessary to have onClick written out here (as it is also covered by the spreaded `props`)?

**Review by mjcarsjens — APPROVED** [review]
One comment but it works

File: components/Button/src/index.tsx (hunk 1 above)
Comment (mjcarsjens) [reply components/Button/src/index.tsx:51]: Also feel a bit weirdly about the `className` being set by the `props` and then overwritten by specifying it to the `className` const, but guess it works

Comment (bddjong) [reply components/Button/src/index.tsx]: Fixed

**Comment by bddjong** [comment]
`type` is now defaulted to `button`. However, all snapshots tests with a button are failing because of this change.

**Pull request merged**